packages/ui/src/components/select.tsx

```diff
@@ -9,38 +9,52 @@ import React, {
     type ChangeEvent,
     useEffect,
 } from "react";
+import AutoSizer from "react-virtualized-auto-sizer";
 import type { BaseInputProps } from "./commons/input";
 import ChevronUp from "../icons/chevron-up";
 import ChevronDown from "../icons/chevron-down";
 import { Popover } from "./popover";
 import { TextInput } from "./text-input";
 import { useClickAway, useDebounce } from "react-use";
 import { mergedCva, mergedCx } from "../utils/components";
+import { FixedSizeList } from "react-window";
 
-const dropdownRootStyles = mergedCva([
-    "cui-rounded-xxl",
-    "cui-border",
-    "cui-border-black",
-    "cui-max-h-48",
-    "cui-bg-white",
-    "dark:cui-bg-black",
-    "dark:cui-border-white",
-    "cui-z-20",
-    "cui-overflow-hidden",
-]);
+const dropdownRootStyles = mergedCva(
+    [
+        "cui-rounded-xxl",
+        "cui-border",
+        "cui-border-black",
+        "cui-max-h-48",
+        "cui-bg-white",
+        "dark:cui-bg-black",
+        "dark:cui-border-white",
+        "cui-z-20",
+        "cui-overflow-hidden",
+        "cui-h-full",
+    ],
+    {
+        variants: {
+            empty: {
+                true: ["cui-h-fit"],
+            },
+        },
+    },
+);
 
-const optionListStyles = mergedCva([
-    "cui-max-h-48",
-    "cui-w-full",
-    "cui-scrollbar",
-    "cui-overflow-y-auto",
+const emptyOptionListStyles = mergedCva([
+    "cui-flex",
+    "cui-justify-center",
+    "cui-items-center",
+    "cui-p-3",
+    "cui-h-12",
 ]);
 
+const optionListStyles = mergedCva(["cui-scrollbar"]);
+
 const optionStyles = mergedCva(
     [
         "cui-cursor-pointer",
         "cui-p-3",
-        "cui-h-12",
```

Comment:
Why remove the `h-12`?

Reply:
Fixed in 3ed97c23e9d9e240a6ab53d46e81013989c7c9f6.